internal/schema/schema.go

```diff
@@ -73,93 +79,104 @@ func (NopManager) ValidatePlanResourcesInput(_ context.Context, _ *policyv1.Sche
 	return alwaysValidResult, nil
 }
 
-func (NopManager) CheckSchema(_ context.Context, _ string) error {
-	return nil
-}
-
-type manager struct {
-	conf     *Conf
-	log      *zap.Logger
-	cache    *cache.Cache[string, *cacheEntry]
-	resolver Resolver
+func (NopManager) LoadSchema(_ context.Context, _ string) (*jsonschema.Schema, error) {
+	return nil, nil
 }
 
-func New(ctx context.Context, loader Loader) (Manager, error) {
+func NewStatic(schemas map[uint64]*policyv1.Schemas, rawSchemas map[string]*runtimev1.RuleTable_JSONSchema) (Manager, error) {
 	conf, err := GetConf()
 	if err != nil {
 		return nil, fmt.Errorf("failed to get config section %q: %w", confKey, err)
 	}
 
-	return NewFromConf(ctx, loader, conf), nil
+	if conf.Enforcement == EnforcementNone {
+		return &StaticManager{}, nil
+	}
+
+	return NewStaticFromConf(conf, schemas, rawSchemas)
 }
 
-func NewFromConf(_ context.Context, loader Loader, conf *Conf) Manager {
+func NewStaticFromConf(conf *Conf, schemas map[uint64]*policyv1.Schemas, rawSchemas map[string]*runtimev1.RuleTable_JSONSchema) (Manager, error) {
 	if conf.Enforcement == EnforcementNone {
-		return NopManager{}
+		return NopManager{}, nil
 	}
 
-	mgr := &manager{
-		conf:     conf,
-		log:      zap.L().Named("schema"),
-		cache:    cache.New[string, *cacheEntry]("schema", conf.CacheSize),
-		resolver: defaultResolver(loader),
+	sm := &StaticManager{
+		conf: conf,
+		log:  zap.L().Named("schema"),
 	}
+	sm.loader = sm
```

Comment:
I'm not super happy with this, it feels like I'm trying to fudge in OOP-like behaviour. That said, I'm not sure how else I should do it 🤷‍♂️